paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -653,7 +653,7 @@ protected List<String> listTablesInFileSystem(Path databasePath) throws IOExcept
     }
 
     protected boolean tableExistsInFileSystem(Path tablePath, String branchName) {
-        return !new SchemaManager(fileIO, tablePath, branchName).listAllIds().isEmpty();
+        return new SchemaManager(fileIO, tablePath, branchName).schemaExists(0);
```

Comment:
if schem-0 not exists, still use `!listAllIds().isEmpty()` to check the table.

Reply:
Updated. By the way, in what scenario would `schema-0` will be deleted.

Reply:
I don't know, but by design, old useless schema files can be deleted.

Reply:
> I don't know, but by design, old useless schema files can be deleted.

ok.